crates/core/tedge_api/src/entity_store.rs

```diff
@@ -1123,9 +1074,25 @@ impl TryFrom<&MqttMessage> for EntityRegistrationMessage {
     type Error = serde_json::Error;
 
     fn try_from(value: &MqttMessage) -> Result<Self, Self::Error> {
-        EntityRegistrationMessage::new(value).ok_or(serde::de::Error::custom(
-            "Failed to parse entity registration message: {value}",
-        ))
+        EntityRegistrationMessage::new(value)
+    }
+}
+
+impl TryFrom<EntityRegistrationPayload> for EntityRegistrationMessage {
+    type Error = serde_json::Error;
+
+    fn try_from(value: EntityRegistrationPayload) -> Result<Self, Self::Error> {
```

Comment:
Implementing this trait makes things more complicated than necessary, while used in a single place.

I would:
- remove the `topic_id` field from `EntityRegistrationPayload`
- implement an ad-hoc method `EntityRegistrationMessage::try_from(TopicId, EntityRegistrationPayload)`

Reply:
Resolved by [8654090](https://github.com/thin-edge/thin-edge.io/pull/3587/commits/8654090f76cf664e56dff2f493927e9a1b324c18)